### ISSUE TITLE
Evaluation

### ISSUE THREAD

**Issue by jubueche** [rajarsheem/libsdae#13]
Nice library.
How did you evaluate the goodness of your feature representation? Is the global loss the general reconstruction?
So for example, the 2nd DAE gets the encoded version of the image, so is the loss function of the 2nd DAE the followin: LOSS(decoded(noisy encoded input), decoded(encoded output))
This way the network doesn't loose seight of the global goal: Finding an optimal feature representation, right?

What I did was: 
1) Train first DAE, get encoded versions of the training data
2) Use the encoded version to  train another DAE with LOSS(noisy encoded, encoded)
I think it was my mistake that I don't consider the completely reconstructed image as the loss..

Is that right?

**Comment by rajarsheem** [comment]
You train the second DAE with the first DAE's encoded as input.
Second DAE tries to reconstruct the first DAE's encoded by performing it's own encoding followed by decoding.

Your second DAE's loss is therefore LOSS(second DAE's decoded, 1st DAE's encoded)
